### PR TITLE
Bright Spot CI blog article

### PR DESCRIPTION
**Pull request by bernhold**
Initial transfer from GDoc

# Instructions (DELETE THIS SECTION)

To view these instructions, please hit the **Preview** tab (above left). After you have read all of these instructions, delete them entirely from the body of this issue before entering your submission.

Be sure to select `master` as the `base` branch above as the target for this PR.

Provide a general summary of your changes in the title above.

Fill in the below **Description** section with minimal text describing the changes/new contributions in this PR and replace `<...>` as appropriate.

Remove the beginning `<!--` and ending `-->` comment markers for the correct **checklist** to be used for this PR and delete the other checklist.  (The checklist items for files displayed on the bssw.io site are shown by default.  The checklist items for internal files not displayed in the bssw.io site is given below with commented out checklist items by default.)

Any checklist items that don't apply can be striked out by adding `~~` to the beginning and end of the checklist item as `* ~~[] <checklist-item>~~`.  Also, remove the strikeout markers `~~` for the [wikize_refs.py] checklist items if using formal citations for bssw.io contributions.

# Description

EB Member: @`<eb-member-id>`

Resolves #`<issue-id>`

`<Other minimal information about the PR.>`

## PR checklist for files displayed on bssw.io site

* [ ] ***[Author]*** `@mention` the BSSw.io editorial board member `@<eb-member-id>` in **Description** above assigned to shepherd your PR.
* [ ] ***[Author]*** Add the `<issue-id>` in the **Description** above for the associated GitHub Issue.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py])~~.
* [ ] ***[Author]*** Inspect the content in the `*.md` file(s) as rendered in GitHub for this PR.
* [ ] ***[EB Mem]*** Assign this PR to the EB member `<eb-member-id>`.
* [ ] ***[EB Mem]*** Assign this PR to the author of the PR `<pr-author-id>`.
* [ ] ***[EB Mem]*** Add one or more Reviewers.
* [ ] ***[EB Mem]*** Add label `content: <content-type>` for the type of contribution.
* [ ] ***[EB Mem]*** Add to Project `Content Development` (see [Content Development]).
* [ ] ***[EB Mem]*** Add [meta-data] to the `*.md` file(s) (set `Publish: yes`).
* [ ] ***[EB Mem]*** Add label `preview` (so PR branch will be merged to 'preview' branch and watch for possible merge failures).
* [ ] ***[EB Mem]*** Rebuild [preview] site and confirm new content is there, renders correctly and is returned in searches.
* [ ] ***[Author]*** Make any final changes to the PR based on feedback.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py]).~~
* [ ] ***[EB Mem]*** Rebuild [preview] site and re-confirm content looks correct.
* [ ] ***[EB Mem]*** Ensure at least one reviewer signs off on the final changes.
* [ ] ***[EB Mem]*** Move the PR to "Ready to Publish" in [Content Development].
* [ ] ***[EB Mem]*** Leave comment and @mention `@betterscientificsoftware/bssw-maint` (**BSSw Maint**) asking to carry out final publication steps.
* [ ] ***[BSSw Maint]*** Verify that all needed files are present in the PR (article, images, updates to Site/Homepage.md carousel and/or Site/Announcements/Announcements.md as appropriate).
* [ ] ***[BSSw Maint]*** Merge PR. (Should automatically move to "Done" in [Content Development].)
* [ ] ***[BSSw Maint]*** Verify new contribution shows up on [bssw.io] as expected.

NOTE:
* Checklist items prefixed with ***[Author]*** are expected to be performed by the author of the PR or can be performed by the author.
* Checklist items prefixed with ***[EB Mem]*** must be performed by a [BSSw.io Editorial Board (EB) Member](https://betterscientificsoftware.github.io/bssw.io/bssw_members.html).
* Checklist items prefixed with ***[BSSw Maint]*** must be performed by someone in the `@betterscientificsoftware/bssw-maint` team (hint: type `@`, `b`, `s`, `s`, `w`, `-`, `m`  to auto-complete to `@betterscientificsoftware/bssw-maint`).

## PR checklist for (internal) files not displayed on bssw.io site

*Click "Write" above and remove comment markers to see below checklist items*

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[Content Development]: https://github.com/betterscientificsoftware/bssw.io/projects/3
[BSSw Internal]: https://github.com/betterscientificsoftware/bssw.io/projects/2
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

File: Articles/Blog/2022-08-brightspot-ci.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+- <img src='../../images/Blog_Brightspot.png' />
+
+#### Contributed by [David M. Rogers](https://github.com/frobnitzem)
+
+#### Publication date: August 12, 2022
+
+*This is the first in an occasional series of Bright Spots blog articles, highlighting success stories of Exascale Computing Project software teams implementing and benefitting from software development best practices.  If you'd like to contribute to a future Bright Spot article, please contact the [BSSw Editorial Board](mailto:bssw-editorial@lists.mcs.anl.gov?subject=Bright Spots).*
+
+[Continuous Integration](https://bssw.io/items/what-is-continuous-integration-testing) is an enabling technology that allows code development teams to insert testing directly into their development workflow[2] .  Although there are many advantages to CI, including more stable working code bases, faster test feedback with less distraction, and greater support for community contributions, there are also some pitfalls[3] , like initial adoption and support burden, disruptive changes to team processes, and a potentially disorienting array of competing, overlapping, new technologies.
```
Comment (bernhold) [inline Articles/Blog/2022-08-brightspot-ci.md:13]: @frobnitzem, we need to do something with the [2] and [3].  We discourage the use of references in blog articles, but we can do that if they're really needed.  But think about whether you really need them here.

File: Articles/Blog/2022-08-brightspot-ci.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
... for [Visit](https://github.com/visit-dav/visit)’s executable metadata-server, engine, viewer and controller.  It has dozens of external dependencies for geometry, graphics, file formats, and I/O.  Visit’s code base was hosted on NERSC’s Subversion servers until early 2019 when it was [migrated](https://bssw.io/blog_posts/continuous-technology-refreshment-an-introduction-using-recent-tech-refresh-experiences-on-visit) to git and GitHub. [Cyrus Harrison](https://github.com/cyrush) first added CI to the codebase around March 2019. Prior to that, only [nightly testing](https://visit-dav.github.io/visit-website/dashboard/) was in use. Adding CI testing was complicated because of the graphical nature of the code and the large number of dependencies.  He opted to use docker images to provide pre-built dependency binaries, combined with a set of scripts that allowed running on both Travis and Circle-CI.  The process to develop the scripts took weeks of trial and error.  The team went on to add more features and switch to Azure pipelines in Aug. 2020.  The CI tests run using a vanilla RedHat Linux base image, but the release process also builds images for Centos, Fedora, Debian and Ubuntu.
```
Comment (bernhold) [inline Articles/Blog/2022-08-brightspot-ci.md:35]: @frobnitzem, Cyrus Harrison is the only person named in this article who has a link.  Do we need that?

File: Articles/Blog/2022-08-brightspot-ci.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+Overall, CI adds value by providing more stable working code bases, test feedback without developer intervention, and greater support for community contributions.  As we watch the evolution of CI technologies, there are several features our teams are looking for.  The ability to freeze the state of tests would both reduce redundant tests after small changes, and also allow interaction with these states.  Greater availability of testing hardware and runners would also boost the usefulness of this technology.
+
+### Acknowledgements
+
+We thank the VisIt team members: A, B, C; ExaSMR members: X, Y, Z, …; and .
```
Comment (bernhold) [inline Articles/Blog/2022-08-brightspot-ci.md:73]: @fronbitzem, need to complete this

File: Articles/Blog/2022-08-brightspot-ci.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+### NOTES (TO BE DELETED?)
```
Comment (bernhold) [inline Articles/Blog/2022-08-brightspot-ci.md:75]: @frobnitzem I wasn't sure if there was anything here that you needed.

File: Articles/Blog/2022-08-brightspot-ci.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+-- mostly off-topic stuff I cut from the manuscript --
+
+Forward-looking statements about benefits and potential of CI pipelines - ecosystem delivery (use cases across projects, release notifications / security alerts), greater automation of integration tests, more focused test selection, cross-project science cases, connecting networks of developers, integrating with documentation
+
+Unique story of ExaSMR, Visit, and ZFP
+
+value: numerous occasions where either testing or pull-request reviews have caught errors that would have gone undetected otherwise, code quality improvements
+
+### Author bios
```
Comment (bernhold) [inline Articles/Blog/2022-08-brightspot-ci.md:87]: @fronbitzem Please provide a short bio (~50-100 words).

**Comment by frobnitzem** [comment]
I will check this against the original markdown formatted version:
#1150 which already contains a bio.

**Comment by bernhold** [comment]
@frobnitzem I didn't realize you'd already produced a PR for it.  Let's drop this one and use #1150.  Thanks

**Pull request closed without merging**